java7/java.l

```diff
@@ -98,13 +98,13 @@ instanceof INSTANCEOF
 \|= OREQ
 
 [0-9]+\.[0-9]*([eE][+-]?[0-9]+)?[fFdD]?|\.[0-9]+([eE][+-]?[0-9]+)?[fFdD]?|[0-9]+[eE][+-]?[0-9]+[fFdD]?|[0-9]+[fFdD] FLOATING_POINT_LITERAL
-(0_[0-7]+|0b[01]+|0x[0-9A-Fa-f]+|[0-9]+)[lL]? INTEGER_LITERAL
+(0_[0-7]+|[0-9][0-9_]+[0-9]|0b[01]+|0[xX][0-9A-Fa-f]+|[0-9]+)[lL]? INTEGER_LITERAL
 (true|false) BOOLEAN_LITERAL
 null NULL_LITERAL
-[a-zA-Z_][a-zA-Z0-9_]* IDENTIFIER
+[a-zA-Z_$][a-zA-Z0-9_$]* IDENTIFIER

```

Comment:
Can an identifier start with `$`?

Reply:
According to the spec, yes: https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8

Reply:
I didn't know that!

Reply:
That wasn't the only thing that surprised me about the Java specification.